### PR TITLE
Document new resize overloads taking Kokkos::view_alloc() return type

### PR DESCRIPTION
**Pull request by masterleinad**
Corresponding to https://github.com/kokkos/kokkos/pull/4844.

### CODE REVIEW

File: docs/source/API/core/view/resize.md [hunk 1]
```diff
@@ -104,6 +145,20 @@ void resize(const I& arg_prop, Kokkos::View<T, P...>& v,
   * `layout`: a layout instance containing the new dimensions.
   * `arg_prop`: View constructor property, e.g., `Kokkos::WithoutInitializing`.
 
+* ```c++
+  template <class T, class... P, class... ViewCtorArgs>
+  void resize(const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop,
+         Kokkos::View<T, P...>& v,
+         const typename Kokkos::View<T, P...>::array_layout& layout);
+  ```
+  Resizes `v` to have the new dimensions while preserving the contents for the common subview of the old and new view. The new `Kokkos::View` is constructed using the View constructor properties `arg_prop`, e.g., `Kokkos::view_alloc(Kokkos::WithoutInitializing)`.
```
Comment (dalg24) [inline docs/source/API/core/view/resize.md:154]: What about use of execution space if provided to copy elements

**Review by dalg24 — COMMENTED** [review]
Did we discuss anywhere whether we really want to support long term both single argument and bundle of them via `view_alloc`?

File: docs/source/API/core/view/resize.md (hunk 1 above)
Comment (masterleinad) [reply docs/source/API/core/view/resize.md:154]: I added a sentence to both overloads. What I really want to say, though, is that providing an execution space follows standard semantics, i.e., use it as much as possible internally and enqueue operations without explicitly fencing. I'm not quite sure if that needs extra documentation and if so where a good place at a central location for such a statement would be.

**Review by crtrott — APPROVED** [review]
I don't like how some of this exposes implementation details, but a lot of it already did. So lets merge it and then do a systematic cleanup later.

**Comment by fnrizzi** [comment]
@masterleinad can you please rebase/resync? I will merge it afterward

**Comment by masterleinad** [comment]
> @masterleinad can you please rebase/resync? I will merge it afterward

Rebased.

**Pull request merged**